const-oid/src/lib.rs

```diff
@@ -1,4 +1,4 @@
-#![no_std]
+#![cfg_attr(not(feature = "std"), no_std)]
```

Comment:
Please don't change these. This makes the semantics non-additive. It's done this way deliberately.

I'm not sure what it's breaking in `arbitrary` not to have this but that seems like a poor use of macros if it's depending on e.g. preludes.

Reply:
Without this change, the code expanded by the macro doesn't compile because the `thread_local` macro cannot be found. I'll try to see if I can make a patch on the `arbitrary` side.

Why does this makes the semantics non-additive ? `serde` [does it this way](https://github.com/serde-rs/serde/blob/f41509261e32357e28a6bc6f46f58e98d841e7e8/serde/src/lib.rs#L86). If the feature `std` is disabled, `no_std` is applied, and if the feature `std` is enabled, `no_std` is not applied.

Reply:
Because you no longer have a consistent baseline: with the current approach, a `no_std` profile is always used as a baseline, and then `std` added back with `extern crate`. Really it's `no_std` that's backwards to begin with, but this approach you're using absence of a feature to remove functionality.

Admittedly it's gotten better than before, when libcore was conditionally linked based on the *presence* of `no_std`, leading to ridiculous hacks like `use core as std`.

But really the problem here is `arbitrary` is depending on the `std` prelude, and yes, that should get fixed. (Why does it even need `thread_local` in the macro anyway?)

Reply:
Ok, I get the difference with the prelude now. I made a PR on rust-fuzz/arbitrary#131. I'll revert the `no_std` change and update this PR when the fix is released.